src/operator/keycloak.ts

```diff
@@ -700,3 +724,103 @@ async function manageGroups(connection: KeycloakConnection) {
     console.error('Error in manageGroups: ', error)
   }
 }
+
+async function removeUserGroups(api: any, existingUser: any, teamGroups: string[]) {
+  try {
+    const { body: existingUserGroups } = await api.users.realmUsersIdGroupsGet(keycloakRealm, existingUser.id as string)
+
+    await Promise.all(
+      existingUserGroups.map(async (group) => {
+        if (!teamGroups.includes(group.name)) {
+          await api.users.realmUsersIdGroupsGroupIdDelete(keycloakRealm, existingUser.id as string, group.id as string)
+        }
+      }),
+    )
+  } catch (error) {
+    console.error('Error removing user groups:', error)
+  }
+}
+
+async function addUserGroups(api: any, existingUser: any, teamGroups: string[]) {
+  try {
+    const { body: currentKeycloakGroups } = await api.groups.realmGroupsGet(keycloakRealm)
+    const { body: existingUserGroups } = await api.users.realmUsersIdGroupsGet(keycloakRealm, existingUser.id as string)
+
+    await Promise.all(
+      teamGroups.map(async (teamGroup) => {
+        const existingGroup = existingUserGroups.find((el) => el.name === teamGroup)
+
+        if (!existingGroup) {
+          const groupId = currentKeycloakGroups.find((el) => el.name === teamGroup)?.id
+          if (groupId) {
+            await api.users.realmUsersIdGroupsGroupIdPut(keycloakRealm, existingUser.id as string, groupId as string)
+          }
+        }
+      }),
+    )
+  } catch (error) {
+    console.error('Error adding user groups:', error)
+  }
+}
+
+async function createUpdateUser(api: any, user: any) {
+  const { email, firstName, lastName, isPlatformAdmin, isTeamAdmin, teams, initialPassword } = user
+  const teamGroups = [
```

Comment:
I am missing a bit adhering the KISS principle.
The keycloak task should create users and assign groups. However I see here a lot of abstract concepts from apl-core like: `isTeamAdmin` or `isPlatformAdmin`. 

IMHO this task should be provided with users where each user is defined as follows `email, firstName, lastName, groups`.

Also hardcoding group names in the keycloak task may not be the best approach because operator that is using it will never see what are the exact group names.
Am I missing sth?

Reply:
Yes, you're right. I should consider that it's an operator function. I will modify the function to accept users with a simplified structure, focusing on email, firstName, lastName and groups and also will refactor the code to avoid hardcoding group names within the function. I can pass the group names as part of the user object and we don't need to use `isTeamAdmin` or `isPlatformAdmin` variables in here. @j-zimnowoda

Reply:
Sounds good.

Reply:
Updated and removed hardcoding group names within the function, apl-core will provide user objects with the following fields: email, firstName, lastName, initialPassword and groups